include/ck/tensor_operation/gpu/grid/gridwise_gemm_reduce_xdl_cshuffle_v1.hpp

```diff
@@ -868,33 +868,41 @@ struct GridwiseGemmReduce_k0mk1_k0nk1_mn_xdl_cshuffle_v1
                                                          make_tuple(I0, I0),
                                                          c_reduce_thread_buf);
 
-                    // reduce in VGPR
+                    // reduce D0 in VGPR
                     ThreadwiseReduce_D0::Reduce(c_reduce_thread_buf, d0_thread_buf);
 
-                    static_for<0, mreduce_per_thread, 1>{}([&](auto im) {
-                        static_for<0, nreduce_per_thread, 1>{}([&](auto in) {
-                            constexpr auto offset =
-                                Number<c_reduce_thread_desc_mperblock_nperblock.CalculateOffset(
-                                    make_tuple(im, in))>{};
-
-                            d1_element_op(c_reduce_thread_buf(offset), c_reduce_thread_buf(offset));
-                        });
-                    });
-
-                    ThreadwiseReduce_D1::Reduce(c_reduce_thread_buf, d1_thread_buf);
-
-                    // copy from VGPR to Global
+                    // copy d0 from VGPR to Global
                     d0_reduce_thread_copy_vgpr_to_global.Run(d_reduce_thread_desc_mblock_mperblock,
                                                              make_tuple(I0, I0),
                                                              d0_thread_buf,
                                                              d_grid_desc_mblock_mperblock,
                                                              d0_grid_buf);
 
-                    d1_reduce_thread_copy_vgpr_to_global.Run(d_reduce_thread_desc_mblock_mperblock,
-                                                             make_tuple(I0, I0),
-                                                             d1_thread_buf,
-                                                             d_grid_desc_mblock_mperblock,
-                                                             d1_grid_buf);
+                    // TODO - Support variable amount of d tensor
+                    if(p_d1_grid)
```

Comment:
It's not a good practice to use run time ```if``` here. Please refactor this gridwise operation to support a tuple of pointers (and a tuple of element-wise ops and reduction-ops)

Reply:
ongoing